src/autumnaton.ts

```diff
@@ -0,0 +1,189 @@
+      }
+    );
+    return seasonalItemValue(location) + expectedCollectionValue;
+  }
+}
+
+function seasonalItemValue(location: Location, seasonalOverride?: number): number {
+  // Find the value of the drops based on zone difficulty/type
+  const autumnItems = $items`autumn leaf, AutumnFest ale, autumn breeze, autumn dollar, autumn years wisdom`;
+  const avgValueOfRandomAutumnItem = garboAverageValue(...autumnItems);
+  const autumnMeltables = $items`autumn debris shield, autumn leaf pendant, autumn sweater-weather sweater`;
+  const autumnItem = AutumnAton.getUniques(location)?.item;
+  const seasonalItemDrops = seasonalOverride ?? AutumnAton.seasonalItems();
+  if (autumnItem) {
+    return (
+      (seasonalItemDrops > 1 ? avgValueOfRandomAutumnItem : 0) +
+      (autumnMeltables.includes(autumnItem)
+        ? // If we already have the meltable, then we get a random item, else value at 0
+          availableAmount(autumnItem) > 0
+          ? avgValueOfRandomAutumnItem
+          : 0
+        : garboValue(autumnItem, true))
+    );
+  } else {
+    // If we're in a location without any uniques, we still get cowcatcher items
+    return seasonalItemDrops > 1 ? avgValueOfRandomAutumnItem : 0;
+  }
+}
+
+function expectedRemainingExpeditions(legOverride?: number): number {
+  const availableAutumnatonTurns = estimatedTurns() - AutumnAton.turnsLeft();
+  let expeditionTurnSum = 0;
+  let quests = get("_autumnatonQuests", 0);
+  while (expeditionTurnSum < availableAutumnatonTurns) {
+    expeditionTurnSum +=
+      11 *
+      Math.max(
+        1,
+        quests -
+          (legOverride ?? AutumnAton.currentUpgrades().filter((u) => u.includes("leg")).length)
+      );
+    quests++;
+  }
+
+  return quests - get("_autumnatonQuests", 0);
+}
+
+const profitRelevantUpgrades = [
+  "leftarm1",
+  "leftleg1",
+  "rightarm1",
+  "rightleg1",
+  "cowcatcher",
+  "periscope",
+  "radardish",
+] as const;
+
+export function mostValuableUpgrade(fullLocations: Location[]): Location[] {
+  // This function shouldn't be getting called if we don't have an expedition left
+  if (expectedRemainingExpeditions() < 1) {
+    return fullLocations;
+  }
+  const currentUpgrades = AutumnAton.currentUpgrades();
+  const acquirableUpgrades = profitRelevantUpgrades.filter(
+    (upgrade) => !currentUpgrades.includes(upgrade)
+  );
+
+  if (acquirableUpgrades.length === 0) {
+    return fullLocations;
+  }
+
+  const currentBestLocation = maxBy(fullLocations, (loc: Location) => averageAutumnatonValue(loc));
+  const currentExpectedProfit =
+    averageAutumnatonValue(currentBestLocation) * expectedRemainingExpeditions();
+
+  const upgradeValuations = acquirableUpgrades.map((upgrade) => {
+    const upgradeLocations = fullLocations.filter(
+      (location) => AutumnAton.getUniques(location)?.upgrade === upgrade
+    );
+    const bestLocContainingUpg = maxBy(upgradeLocations, (loc: Location) =>
+      averageAutumnatonValue(loc)
+    );
```

Comment:
```suggestion
    const bestLocContainingUpgrade = maxBy(upgradeLocations, averageAutumnatonValue);
```

constructs of the form `(x) => f(x)` are just copying the function `f` in-line. Sometimes that's useful for legibility, but imo it is nicer to read the other way.

Reply:
Also, letters don't cost us anything; we should use the full word! "upgrade" (that change means that committing this will mean manually fixing some stuff that still references the old name)